### PR TITLE
Implement mongo tests

### PR DESCRIPTION
**Pull request by joshfttb**

### CODE REVIEW

File: test/jMongoTests.js [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+        });
+    });
+
+    describe( 'verify', () => {
+        // userId, path, operation
+        const rejectUser = mongoose.Types.ObjectId();
+        it( 'should allow reading a file with correct permissions', () => {
+            expect( mongo.verify( userId, '/level1/level2/level3/test.txt', 'read' )).to.be.fulfilled();
+        });
+        it( 'should reject reading a file with correct permissions', () => {
```
Comment (JustinLivi) [inline test/jMongoTests.js]: this should be "with incorrect permissions"

File: test/jMongoTests.js [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+            expect( mongo.verify( rejectUser, '/level1/level2/level3/test.txt', 'read' ))
+                .to.be.rejectedWith( 'user does not have read permissions on this object' );
+        });
+        it( 'should allow updating a file with correct permissions', () => {
+            expect( mongo.verify( userId, '/level1/level2/level3/test.txt', 'update' )).to.be.fulfilled();
+        });
+        it( 'should reject updating a file with correct permissions', () => {
```
Comment (JustinLivi) [inline test/jMongoTests.js]: with incorrect permissions

File: test/jMongoTests.js (hunk 1 above)
Comment (JustinLivi) [inline test/jMongoTests.js]: Actually nevermind, these should be in the permissions module

**Comment by JustinLivi** [comment]
Once we remove the generators from this I'll merge

**Comment by BrianSilvia** [comment]
The tests are going to have to be restructured a bit. We want to test the mongo operations themselves -- insert, update, search, delete, not the API actions. We also want to make sure that what we're testing falls in line with the end-to-end API that we now have

**Comment by JustinLivi** [comment]
After talking with Brian a lot of refactoring needs to happen so I'm closing this for now.

**Pull request closed without merging**